src/core/reducers/user.js

```diff
@@ -0,0 +1,51 @@
+/* @flow */
+import { LOG_OUT_USER } from 'core/constants';
+
+
+const LOAD_USER_PROFILE = 'LOAD_USER_PROFILE';
+
+type Action = Object;
+export type UserStateType = {
+  id: ?number,
+  username: ?string,
+};
+
+export const initialState: UserStateType = {
+  id: null,
+  username: null,
+};
+
+export const loadUserProfile = ({ profile }: Object) => {
+  if (!profile) {
+    throw new Error('The profile parameter is required.');
+  }
+
+  return {
+    type: LOAD_USER_PROFILE,
+    payload: { profile },
+  };
+};
+
+export const isAuthenticated = (state: { user: UserStateType }) => {
+  return !!state.user.id;
+};
+
+export default function reducer(
+  state: UserStateType = initialState,
+  action: Action = {}
```

Comment:
for future reference, this would look more like:

```js
action: LoadUserProfileAction | LogOutUserAction
```

...but we'd have to switch away from constants to strings so that Flow [can figure out which action is which](https://flow.org/en/docs/react/redux/#toc-typing-redux-reducers) in the switch.